lib/plugins/filter/post_permalink.js

```diff
@@ -42,7 +43,15 @@ function postPermalinkFilter(data) {
     Object.defineProperty(meta, key, Object.getOwnPropertyDescriptor(data, key));
   }
 
-  return permalink.stringify(Object.assign(meta, config.permalink_defaults));
+  const keys2 = Object.keys(config.permalink_defaults);
+
+  for (const key of keys2) {
+    if (Object.prototype.hasOwnProperty.call(meta, key)) continue;
+
+    meta[key] = config.permalink_defaults[key];
```

Comment:
Try [`deepMerge()`](https://github.com/hexojs/hexo-util#deepmergetarget-source) to see if `Maximum call stack size exceeded` still occurs.